### PR TITLE
[DOCS] Getting started with ES|QL queries guide

### PR DESCRIPTION
**Pull request by abdonpijpelink**
This adds a "getting started with ES|QL queries guide". It guides users from a simple FROM, to adding processing commands, chaining processing commands, calculating statistics with STATS..BY, data enrichment with ENRICH, and data processing with DISSECT/GROK.

Preview: https://elasticsearch_101634.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql-getting-started.html

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
@elasticmachine test this please

File: docs/reference/esql/esql-get-started.asciidoc [hunk 1]
```diff
@@ -1,34 +1,434 @@
...
+
+After creating and executing a policy, you can use it with the `ENRICH`
+command:
+
+[source,esql]
+----
+FROM sample_data
+| KEEP @timestamp, client.ip, event.duration
+| EVAL client.ip = TO_STRING(client.ip)
+| ENRICH clientip_policy
```
Comment (luigidellaquila) [inline docs/reference/esql/esql-get-started.asciidoc:363]: I'd use the extended ENRICH syntax here, ie `ENRICH policy ON field`, that IMHO is more clear

**Review by luigidellaquila — APPROVED** [review]
LGTM in general, it just needs a little fix (`column` vs `row`, see the comment)

File: docs/reference/esql/esql-get-started.asciidoc [hunk 2]
```diff
@@ -1,34 +1,434 @@
...
+
+NOTE: The order of processing commands is important. First limiting the result
+set to 3 rows before sorting those 3 rows would most likely return 3 different
+rows.
+
+[discrete]
+[[esql-getting-started-eval]]
+=== Compute values
+
+Use the <<esql-eval>> command to append rows to a table, with calculated values.
```
[the remark this answers is not in the document]
Comment (abdonpijpelink) [reply docs/reference/esql/esql-get-started.asciidoc:201]: Good catch! Thanks.

File: docs/reference/esql/esql-get-started.asciidoc [hunk 3]
```diff
@@ -1,34 +1,434 @@
...
-== Getting started with {esql}
+== Getting started with {esql} queries
 
 ++++
 <titleabbrev>Getting started</titleabbrev>
 ++++
 
-A simple example of an {esql} query is shown below:
-[source,esql]
+This guide shows you can use {esql} to query and aggregate your data.
```
Comment (astefan) [inline docs/reference/esql/esql-get-started.asciidoc:8]:
```suggestion
This guide shows how you can use {esql} to query and aggregate your data.
```

File: docs/reference/esql/esql-get-started.asciidoc [hunk 4]
```diff
@@ -1,34 +1,434 @@
...
+
+[source,esql]
+----
+FROM sample_data
+| SORT @timestamp DESC
+| LIMIT 3
+----
+
+NOTE: The order of processing commands is important. First limiting the result
+set to 3 rows before sorting those 3 rows would most likely return 3 different
```
Comment (astefan) [inline docs/reference/esql/esql-get-started.asciidoc:194]: Not sure what you meant here: is that the result of `sort | limit` is different than the one of `limit | sort`? If so the statement " First limiting the result
set to 3 rows before sorting those 3 rows would most likely return 3 different rows" would be better off like " First limiting the result set to 3 rows before sorting those 3 rows would most likely return a result that is different than the one where the sorting comes before the limit".

File: docs/reference/esql/esql-get-started.asciidoc [hunk 5]
```diff
@@ -1,34 +1,434 @@
...
+
+You can calculate multiple stats with one command:
+
+[source,esql]
+----
+FROM sample_data
+| STATS median_duration = MEDIAN(event.duration), max_duration = MAX(event.duration)
+----
+
+Use `BY` to group calculated stats by one or more other values. For example, to
```
Comment (astefan) [inline docs/reference/esql/esql-get-started.asciidoc:244]:
```suggestion
Use `BY` to group calculated stats by one or more columns. For example, to
```

File: docs/reference/esql/esql-get-started.asciidoc [hunk 6]
```diff
@@ -1,34 +1,434 @@
...
+FROM sample_data
+| STATS median_duration = MEDIAN(event.duration) BY client.ip
+----
+
+[discrete]
+[[esql-getting-started-histogram]]
+=== Create a histogram
+
+To track statistics over time, {esql} enables you to create histograms using the
+<<esql-auto_bucket>> function. `AUTO_BUCKET` creates human-friendly buckets and
```
Comment (astefan) [inline docs/reference/esql/esql-get-started.asciidoc:258]:
```suggestion
<<esql-auto_bucket>> function. `AUTO_BUCKET` creates human-friendly bucket sizes and
```

**Review by astefan — APPROVED** [review]
LGTM with few small comments. Thanks @abdonpijpelink, these look great.

File: docs/reference/esql/esql-get-started.asciidoc (hunk 4 above)
Comment (abdonpijpelink) [reply docs/reference/esql/esql-get-started.asciidoc]: Yes, that's what I was trying to say here. Thanks! I've changed it into your suggestion.

**Pull request merged**